facebook/OpenBIC
# op2: op: Initialize Olmsted Point BIC

Summary:
- Initialize Olmsted Point BIC.
- Because Olympus 2.0 server board can install two Olmsted Point, Olmsted Point is classified into two types. One is "Olmsted Point A" and the other is "Olmsted Point B".
- Add project code in meta-facebook/op2-opa/ and meta-facebook/op2-opb/.
- Version commit for Expander BIC A op2-opa-2022.49.01.
- Version commit for Expander BIC B op2-opb-2022.49.01.

Test Plan:
- Build code: PASS
- Access to BIC console(UART): PASS
- Check GPIO list: PASS

Test Log:
    1. build code :
    ```
    [276/276] Linking C executable zephyr/OP2BOPB.elf
    Memory region         Used Size  Region Size  %age Used
         SRAM_NC:         12 KB       320 KB      3.75%
           FLASH:          0 GB         0 GB
            SRAM:      256016 B       448 KB     55.81%
        IDT_LIST:          0 GB         2 KB      0.00%

    2. Access to BIC console :

    2-1. Olmsted Point A:

    uart:~$ platform info
    ========================{SHELL COMMAND INFO}========================================
    * NAME:          Platform command
    * DESCRIPTION:   Commands that could be used to debug or validate.
    * DATE/VERSION:  none
    * CHIP/OS:       ast1030_evb - Zephyr
    * Note:          none
    ------------------------------------------------------------------
    * PLATFORM:      Olympic 2.0-Olmsted Point A
    * BOARD ID:      1
    * STAGE:         0
    * SYSTEM:        255
    * FW VERSION:    1.0
    * FW DATE:       2022.41.1
    * FW IMAGE:      OP2BOPA.bin
    ========================{SHELL COMMAND INFO}========================================
    uart:~$

    2-2. Olmsted Point B:

    root@bmc-oob:~# /usr/local/bin/mTerm_client slot2
    ```
    uart:~$ platform info
    ========================{SHELL COMMAND INFO}========================================
    * NAME:          Platform command
    * DESCRIPTION:   Commands that could be used to debug or validate.
    * DATE/VERSION:  none
    * CHIP/OS:       ast1030_evb - Zephyr
    * Note:          none
    ------------------------------------------------------------------
    * PLATFORM:      Olympic 2.0-Olmsted Point B
    * BOARD ID:      1
    * STAGE:         0
    * SYSTEM:        1
    * FW VERSION:    1.0
    * FW DATE:       2022.41.1
    * FW IMAGE:      OP2BOPB.bin
    ========================{SHELL COMMAND INFO}========================================
    uart:~$

    3. Check GPIO list :

    3-1. Olmsted Point A:
    uart:~$ platform gpio list_all
    [0  ] CLKBUF_E1S_0_OE_N                  : PP  | output(O) | 1(1)
    [1  ] CLKBUF_E1S_1_OE_N                  : PP  | output(O) | 1(1)
    [2  ] CLKBUF_E1S_2_OE_N                  : PP  | output(O) | 1(1)
    ```
    [37 ] E1S_2_12V_POWER_EN                 : PP  | output(O) | 0(0)
    [38 ] EN_P0V9_VR                         : PP  | output(O) | 0(0)

    3-2. Olmsted Point B:
    uart:~$ platform gpio list_all
    [0  ] CLKBUF_E1S_0_OE_N                  : PP  | output(O) | 1(1)
    [1  ] CLKBUF_E1S_1_OE_N                  : PP  | output(O) | 1(1)
    [2  ] CLKBUF_E1S_2_OE_N                  : PP  | output(O) | 1(1)
    ```
    [38 ] E1S_3_POWER_EN                     : PP  | output(O) | 0(0)
    [39 ] E1S_4_POWER_EN                     : PP  | output(I) | 0(0)

Note:
- GPIO table version : 20221114
